fmc_policies.tf

```
@@ -118,6 +118,13 @@ locals {
                 type = local.map_endpoint_device_types[endpoint_device_type].type
                 name = local.map_endpoint_device_types[endpoint_device_type].name
               }]
+
+              applications = [for application in try(rule.applications, []) : {
+                name = application
+                id   = try(data.fmc_applications.module[domain.name].items[application].id, null)
+                type = try(data.fmc_applications.module[domain.name].items[application].type, "Application")
```

```suggestion
id   = try(data.fmc_applications.module[domain.name].items[application].id, null)
```

Since provider for [rules.applications](https://registry.terraform.io/providers/CiscoDevNet/fmc/latest/docs/resources/access_control_policy#nested-schema-for-rulesapplications) defines only `id` field, perhaps it would be better to get rid of remaining ones.